libs/execution_context/ebpf_core.c

```diff
@@ -1065,6 +1071,89 @@ _ebpf_core_protocol_get_next_program_id(
     return _get_next_id(EBPF_OBJECT_PROGRAM, request, reply, reply_length);
 }
 
+// Get standard cross-platform bpf_link_info for a link object.
+static ebpf_result_t
+_ebpf_core_protocol_get_link_object_info(
+    ebpf_handle_t handle, _Out_writes_to_(*info_size, *info_size) uint8_t* buffer, _Inout_ uint16_t* info_size)
+{
+    ebpf_link_t* link;
+    ebpf_result_t result = ebpf_reference_object_by_handle(handle, EBPF_OBJECT_LINK, (ebpf_object_t**)&link);
+    if (result != EBPF_SUCCESS) {
+        return result;
+    }
+    result = ebpf_link_get_info(link, buffer, info_size);
+    ebpf_object_release_reference((ebpf_object_t*)link);
+    return result;
+}
+
+// Get standard cross-platform bpf_map_info for a map object.
+// Compare _ebpf_core_protocol_get_map_info which is used to get
+// map pinning info.
+static ebpf_result_t
+_ebpf_core_protocol_get_map_object_info(
+    ebpf_handle_t handle, _Out_writes_to_(*info_size, *info_size) uint8_t* buffer, _Inout_ uint16_t* info_size)
+{
+    ebpf_map_t* map;
+    ebpf_result_t result = ebpf_reference_object_by_handle(handle, EBPF_OBJECT_MAP, (ebpf_object_t**)&map);
+    if (result != EBPF_SUCCESS) {
+        return result;
+    }
+    result = ebpf_map_get_info(map, buffer, info_size);
+    ebpf_object_release_reference((ebpf_object_t*)map);
+    return result;
+}
+
+// Get standard cross-platform bpf_prog_info for a program object.
+// Compare _ebpf_core_protocol_get_program_info which is used
+// by the verifier/jitter to get helper info for a program or
+// program type.
+static ebpf_result_t
+_ebpf_core_protocol_get_program_object_info(
+    ebpf_handle_t handle, _Out_writes_to_(*info_size, *info_size) uint8_t* buffer, _Inout_ uint16_t* info_size)
+{
+    ebpf_program_t* program;
+    ebpf_result_t result = ebpf_reference_object_by_handle(handle, EBPF_OBJECT_PROGRAM, (ebpf_object_t**)&program);
+    if (result != EBPF_SUCCESS) {
+        return result;
+    }
+    result = ebpf_program_get_info(program, buffer, info_size);
+    ebpf_object_release_reference((ebpf_object_t*)program);
+    return result;
+}
+
+static ebpf_result_t
+_ebpf_core_protocol_get_object_info(
+    _In_ const ebpf_operation_get_object_info_request_t* request,
+    _Out_ ebpf_operation_get_object_info_reply_t* reply,
+    uint16_t reply_length)
+{
+    uint16_t info_size = reply_length - FIELD_OFFSET(ebpf_operation_get_object_info_reply_t, info);
+    ebpf_result_t result;
+
+    // Try a link handle.
```

Comment:
Instead of trying for each type, we can do the following also:

1. Call `ebpf_reference_object_by_handle()` and pass `EBPF_OBJECT_UNKNOWN`
2. Get object pointer and check type
3. based on the type, call the corresponding `_get_info()` function

Reply:
Good point, done